apple/internal/transition_support.bzl

```diff
@@ -102,6 +102,9 @@ def _cpu_string(*, cpu, platform_type, settings = {}):
             return "ios_sim_arm64"
         return "ios_x86_64"
     if platform_type == "macos":
+        host_cpu = settings["//command_line_option:host_cpu"]
+        if host_cpu == "darwin":
+            return "darwin"
```

Comment:
this should be below all the other conditions in this block, since --cpu and --macos_cpus should take precedence over this logic

Reply:
I tried moving it below the `cpu` and `macos_cpu` options and both failed

Reply:
Doesn't this order mean users couldn't cross compile things then for darwin_arm64?

Reply:
right but if you are on your intel mac, where the host_cpu is `darwin`, and you pass `--cpu=darwin_arm64` to produce an output for an arm64 mac, i think this condition would take precedence

Reply:
I can confirm that this change does break apple silicone mac. I'm also facing another [problem](https://github.com/bazelbuild/rules_foreign_cc/issues/1051) with building on apple silicone. Is there a better approach that could be used to solve this problem without breaking things for the above condition?

Reply:
you'd probably need to check the times we return darwin_x86_64 and return `darwin` in the case that the host CPU is just `darwin`, and not affect the return  value otherwise